plugins/alerting/pkg/alerting/ops/reconcile.go

```diff
@@ -3,17 +3,26 @@ package ops
 import (
 	"context"
 	"fmt"
+	"net/url"
+	"path"
 	"time"
 
+	"github.com/rancher/opni/pkg/alerting/shared"
 	"github.com/rancher/opni/pkg/alerting/storage"
+	sync_opts "github.com/rancher/opni/pkg/alerting/storage/opts"
 	alertingv1 "github.com/rancher/opni/pkg/apis/alerting/v1"
+	"github.com/rancher/opni/pkg/util"
 	"github.com/rancher/opni/plugins/alerting/pkg/apis/alertops"
 	"google.golang.org/grpc/codes"
 	status "google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/emptypb"
 	"gopkg.in/yaml.v2"
 )
 
+func cacheEndpointToWebhook(endpoint string) string {
+	return fmt.Sprintf("http://%s", path.Join(endpoint, shared.AlertingDefaultHookName))
```

Comment:
this seems error-prone. also why http?

Reply:
this is stop-gap solution to vanilla AlertManager only exposing cluster-peer addresses in its memberlist api, https://github.com/rancher/opni/issues/1150 will address these error prone definitions